packages/components/src/ui/utils/space.ts

```diff
@@ -1,16 +1,44 @@
 /**
- * External dependencies
+ * A real number or something parsable as a number
  */
-// eslint-disable-next-line no-restricted-imports
-import type { ReactText } from 'react';
+export type SpaceInput = number | string;
+
+const GRID_BASE = '4px';
+
 /**
- * Internal dependencies
+ * A function that handles numbers, numeric strings, and unit values.
+ *
+ * When given a number or a numeric string, it will return the grid-based
+ * value as a factor of GRID_BASE, defined above.
+ *
+ * When given a unit value, it will simply return the unit value back.
+ *
+ * @param  value A number, numeric string, or a unit value.
  */
+export function space( value?: SpaceInput ): string | undefined {
+	if ( typeof value === 'undefined' ) {
+		return undefined;
+	}
 
-const GRID_BASE = '4px';
+	// handle empty strings, if it's the number 0 this still works
+	if ( ! value ) {
+		return '0';
+	}
+
+	// test if the input has a unit, in which case just use that value
+	if (
+		typeof CSS.supports === 'function' &&
+		CSS.supports( 'margin', value.toString() )
```

Comment:
Can this be rewritten as 

```suggestion
		CSS?.supports?.( 'margin', value.toString() )
```

Reply:
We could also further simplify the code in this function by checking for `CSS?.supports?.( 'margin', value.toString() ) || Number.isNaN( asInt )) in the same `if` block, since in both cases we are returning `value.toString();`